client/src/components/UndoRedo/UndoRedoStack.vue

```diff
@@ -0,0 +1,152 @@
+<script setup lang="ts">
+import { library } from "@fortawesome/fontawesome-svg-core";
+import { faHistory } from "@fortawesome/free-solid-svg-icons";
+import { ref, watch } from "vue";
+
+import { useUndoRedoStore } from "@/stores/undoRedoStore";
+
+import Heading from "@/components/Common/Heading.vue";
+
+library.add(faHistory);
+
+const props = defineProps<{
+    storeId: string;
+}>();
+
+const currentStore = ref(useUndoRedoStore(props.storeId));
+
+watch(
+    () => props.storeId,
+    (id) => (currentStore.value = useUndoRedoStore(id))
+);
+</script>
+
+<template>
+    <section class="undo-redo-stack">
+        <Heading h2 size="sm" icon="fa-history">Latest Changes</Heading>
+
+        <div class="scroll-list">
+            <button
+                v-for="action in currentStore.redoActionStack"
+                :key="action.id"
+                class="action future"
+                @click="currentStore.rollForwardTo(action)">
+                {{ action.name }}
+            </button>
+
+            <span class="state-indicator"> current state </span>
+
+            <button v-if="currentStore.pendingLazyAction" class="action lazy" @click="currentStore.undo">
+                {{ currentStore.pendingLazyAction.name }}
+            </button>
+
+            <button
+                v-for="action in [...currentStore.undoActionStack].reverse()"
+                :key="action.id"
+                class="action past"
+                @click="currentStore.rollBackTo(action)">
+                {{ action.name }}
+            </button>
+
+            <span v-if="currentStore.deletedActions.length !== 0" class="state-indicator"> latest saved state </span>
+
+            <span v-for="(action, i) in [...currentStore.deletedActions].reverse()" :key="i" class="action dead">
+                {{ action }}
+            </span>
+
+            <span class="state-indicator"> start of session </span>
+        </div>
+
+        <label>
+            Max saved changes
```

Comment:
The minimum of 10 does not seem to be enforced, I can set this lower and it works fine.

Reply:
I understand that part, but since the _minimum_ is defined as 10 and I can set this to 5 I reported this as a tiny discrepancy.

Reply:
Ah, I misread